boa_engine/src/context/mod.rs

```diff
@@ -202,6 +205,15 @@ impl Context<'_> {
         result
     }
 
+    /// Applies optimizations to the [`StatementList`] inplace.
+    pub fn optimize_statement_list(&mut self, statement_list: &mut StatementList) {
+        let mut optimizer = Optimizer::new(self);
+        let mut cfo = ConstantFoldingOptimizer::default();
+        optimizer.push_pass(&mut cfo);
```

Comment:
Maybe it would be better to have `ConstantFoldingOptimizer` inside `Optimizer`, since it abstracts the `optimize` action from the optimization types and also makes it easier to add more optimizations in the future.